checks/raw/shell_download_validate.go

```diff
@@ -696,6 +697,83 @@ func isChocoUnpinnedDownload(cmd []string) bool {
 	return true
 }
 
+func isUnpinnedNugetCliInstall(cmd []string) bool {
+	// Search for nuget commands.
+	if !isBinaryName("nuget", cmd[0]) {
+		return false
+	}
+
+	// Search for install commands.
+	if !strings.EqualFold(cmd[1], "install") {
+		return false
+	}
+
+	// package.config schema has required version field
+	// https://learn.microsoft.com/en-us/nuget/reference/packages-config#schema
+	if strings.HasSuffix(cmd[2], "packages.config") {
+		return false
+	}
+
+	hasVersion := false
+	for i := 2; i < len(cmd); i++ {
+		// look for version flag
+		if strings.EqualFold(cmd[i], "-Version") {
+			hasVersion = true
+			break
+		}
+	}
+
+	if hasVersion {
+		return !hasVersion
+	}
+
+	return true
+}
+
+func isUnpinnedDotNetCliInstall(cmd []string) bool {
+	// Search for dotnet commands.
+	if !isBinaryName("dotnet", cmd[0]) {
```

Comment:
nit: add a note that length of `cmd` is verified by the caller. I don't think we've added these commands in the rest of the code, we / I should have :)

Reply:
changed where the check is performed